src/fabric.c

```diff
@@ -463,6 +509,17 @@ void fi_ini(void)
 			" (default: OFI service specific)");
 	fi_param_get_str(NULL, "provider", &param_val);
 	ofi_create_filter(&prov_filter, param_val);
+	ofi_create_filter(&prov_post_filter, param_val);
+
+	/* remove layering from the base filter */
+	if (prov_filter.names) {
+		int i;
+		char *p;
+		for (i=0; prov_filter.names[i]; i++) {
+			if ((p = strchr(prov_filter.names[i], ';')))
+				*p = '\0';
+		}
+	}
```

Comment:
I wonder if we should move the creation of the provider filters to a new call.

Reply:
It was originally that way. I can change it back if that is more preferable.